File: java/389.java
class Solution {
    public char findTheDifference(String s, String t) {
        Map<Character,Integer> map = new HashMap();
        for(char c:s.toCharArray()){
            int k = map.getOrDefault(c,0);
            map.put(c,k+1);
        }
        for(char c:t.toCharArray()){
            int k = map.getOrDefault(c,0);
            if(k == 0){
                return c;
            }
            map.put(c,k-1);
        }
        return 'a';
    }
}

File: go/389.go
func findTheDifference(s string, t string) byte {
	sMap := make(map[byte]int)
	for i := 0; i < len(s); i++ {
		value, ok := sMap[s[i]]
		if ok {
			sMap[s[i]] = value + 1
		} else {
			sMap[s[i]] = 1
		}
	}
	for i := 0; i < len(t); i++ {
		value, ok := sMap[t[i]]
		if !ok {
			return t[i]
		} else {
			if value == 0 {
				return t[i]
			} else {
				sMap[t[i]] = value - 1
			}
		}
	}
	return 'a'
}